lib/common/ipc_client.c

```diff
@@ -1205,34 +1206,38 @@ internal_ipc_get_reply(crm_ipc_t *client, int request_id, int ms_timeout,
             break;
         }
 
-        if (hdr->qb.id == request_id) {
+        reply_id = hdr->qb.id;
+
+        if (reply_id == request_id) {
             /* Got the reply we were expecting. */
-            if (client->buffer != NULL) {
```

Comment:
With this commit, we're now `CRM_CHECK`-ing that `*(client->buffer) == NULL` instead of freeing the buffer. That's probably fine internally, although I haven't traced every caller to be sure the buffer is already `NULL`. We do free it at the end of this function, regardless.

I'm not too worried about public callers, if they're calling this with a nonempty client buffer for some reason.